### PR TITLE
fix possible agent_proxy resource leak

### PR DESCRIPTION
**Pull request by gy95**
Signed-off-by: gy95 <1015105054@qq.com>
fix possible agent_proxy resource leak: where to release resource(close connection or file) should be modified.

### CODE REVIEW

File: agent/pkg/proxy/proxy.go [hunk 1]
```diff
@@ -59,8 +59,10 @@ func realServerAddress(conn *net.Conn) (string, int, error) {
 
 	file, err := tcpConn.File()
 	if err != nil {
+		tcpConn.Close()
```
Comment (ZBoIsHere) [inline agent/pkg/proxy/proxy.go:62]: It doesn't need to close tcpConn here, because this function caller do that thing

Comment (gy95) [reply agent/pkg/proxy/proxy.go:62]: yes, you're right

**Pull request merged**